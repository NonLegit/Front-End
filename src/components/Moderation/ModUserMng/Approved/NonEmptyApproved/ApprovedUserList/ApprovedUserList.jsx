import React from 'react';
import { useParams } from 'react-router-dom';
import { ApprovedFetch } from '../../ApprovedServer';
import ApprovedUser from '../ApprovedUser/ApprovedUser';
import calculateTime from '../../../../../../utils/calculateTime';
import SearchBar from '../../../SearchBar/SearchBar';
import SearchResultBar from '../../../SearchResultBar/SearchResultBar';
import NoResult from '../../../NoResult/NoResult';
import EmptyApproved from '../../EmptyApproved/EmptyApproved';

/**
 * approved user list
 * @component
 * @return {React.Component} - approved user list
 */

function ApprovedUserList() {
  const [data, setData] = React.useState('');
  const childToParent = (childData) => {
    setData(childData);
  };

  const { subReddit } = useParams();
  const [ApprovedUsers] = ApprovedFetch(subReddit);

  const [filteredData, setfilteredData] = React.useState([]);

  React.useEffect(() => {
    setfilteredData(ApprovedUsers.filter((user) => user.user.userName.toLowerCase().includes(data)));
  }, [data, ApprovedUsers]);

  return (
    (ApprovedUsers.length === 0) ? (<EmptyApproved />) : (
      <>
        <SearchBar childToParent={childToParent} />
        {
        (filteredData.length !== 0 && data !== '') && (
        <SearchResultBar
          resultNumber={filteredData.length}
          filteredData={filteredData}
          childToParent={childToParent}
        />
        )
      }
        {
       (filteredData.length === 0 && data !== '') && <NoResult query={data} childToParent={childToParent} />
      }
        {
        filteredData.map((users, index) => {
          const {
            user, approvedDate,
          } = users;
          return (
            <ApprovedUser
              key={`${index + 0}`}
              userName={user.userName}
              profilePicture={user.profilePicture}
              joiningDate={calculateTime(approvedDate)}
            />
          );
        })
      }
      </>
    )
  );
}

export default ApprovedUserList;
