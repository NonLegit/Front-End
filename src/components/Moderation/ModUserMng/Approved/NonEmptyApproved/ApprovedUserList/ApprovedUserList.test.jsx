import renderer from 'react-test-renderer';
import ApprovedUserList from './ApprovedUserList';

test('test snapshot', async () => {
  const tree = renderer.create(<ApprovedUserList />).toJSON();
  expect(tree).toMatchSnapshot();
});
