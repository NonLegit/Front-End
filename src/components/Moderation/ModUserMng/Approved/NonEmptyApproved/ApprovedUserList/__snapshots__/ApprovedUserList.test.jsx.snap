// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-ekpinx"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-zbb00y-MuiSvgIcon-root"
    data-testid="CreateOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM5.92 19H5v-.92l9.06-9.06.92.92L5.92 19zM20.71 5.63l-2.34-2.34c-.2-.2-.45-.29-.71-.29s-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41z"
    />
  </svg>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-155szis-MuiTypography-root"
  >
    No approved users in r/
     
  </h6>
</div>
`;
